pkg/controller/render/render_controller.go

```diff
@@ -423,6 +423,13 @@ func (ctrl *Controller) syncMachineConfigPool(key string) error {
 	if err != nil {
 		return err
 	}
+	requirements, _ := selector.Requirements()
+	values := requirements[0].Values()
+	_, workerExist := values["worker"]
+	_, masterExist := values["master"]
+	if !masterExist && !workerExist {
+		klog.Error("Every custom MCP must inherit from Worker")
+	}
```

Comment:
agreed

Reply:
sorry, not sure I'm following! I think we should be safe to remove this bit since our VAP will hopefully block it at the API level. Is there a reason you are leaving this in?

Reply:
updated